In [1]:
import pandas as pd

from sklearn.model_selection import train_test_split
from sklearn.pipeline import make_pipeline
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.linear_model import LogisticRegression

Read in the data and then split into true and false so we have equal numbers of trues and falses in the train and test.

In [2]:
data = pd.read_json("../data/clean.json")
trues = data[data["label"]]
falses = data[data["label"] == False]

Split the trues and falses into test and train

In [3]:
t_train, t_test = train_test_split(trues, test_size=0.5, random_state=13)
f_train, f_test = train_test_split(falses, test_size=0.5, random_state=13)
train = pd.concat([t_train, f_train])
test = pd.concat([t_test, f_test])

Init the pipeline

In [4]:
pipe = make_pipeline(
    CountVectorizer(),
    LogisticRegression(max_iter=300),
)

Fit and score

In [5]:
pipe.fit(train["content"], train["label"])
pipe.score(test["content"], test["label"])

0.7213622291021672

This is an accuracy of 0.7213622291021672 which is suspiciously high. It gets this every time.